docs/theory.md

```diff
@@ -0,0 +1,333 @@
+
+NAGL solves this problem by training machine learning models to predict partial charges - and potentially other properties - from a **molecular graph**. Machine learning describes a wide array of techniques that can produce  a cheap-to-compute approximation to an arbitrary mathematical function by interpolating between example evaluations of the function in a high-dimensional space. This addresses all three drawbacks:
+
+1. Because the input graph is based on the connectivity of the atoms rather than their co-ordinates, the output is conformation-independent (as long as the training data was properly prepared)
+2. The model can be evaluated much more quickly than a QC calculation, and scales to large molecules efficiently
+3. NAGL can take advantage of the thriving software ecosystem around machine learning, and so can be executed on all sorts of molecules and machines with relative ease
+
+In trade, machine learning models require large datasets of examples of the desired property. As they are at heart interpolations between these examples, the details of the structure of the models and the design of the example datasets are crucial to the production of reliable models. NAGL is designed to make this possible.
+
+## The Molecular Graph
+
+A molecular graph is essentially a Lewis structure represented as a mathematical graph. A  graph is a collection of **nodes** that are connected by **edges**. Graphs can model lots of everyday systems and objects: in computer networks, computers are nodes and WiFi or Ethernet connections are edges; for an airline, airports are nodes and flights are edges; on Twitter, accounts are nodes and follows are edges. 
+
+In a molecular graph, atoms are nodes and bonds are edges. A molecular graph can store extra information in its nodes and edges; in a Lewis structure, this is things like an atom's element and lone pairs, or a bond's bond order or stereochemistry, but for a molecular graph they can be anything. Element and bond order are usually somewhere near the top of the list. Envisioning a molecule as a graph lets us apply computer science techniques designed for graphs to molecules.
+
+Note that a particular molecular species may have more than one molecular graph topology that represents it. This most commonly happens with tautomers and resonance forms. NAGL operates on molecular graphs, and on the OpenFF [`Molecule`] class, and it doesn't try to be clever about whether two molecules are the same or not. If you want tautomers to produce the same charges, you may need to prepare your dataset accordingly.
```

Comment:
(Non-blocking, since the ResonanceEnumerator is not very well documented and hidden in utils) -- this might be a good place to eventually mention the ResonanceEnumerator which will enumerate all resonance forms of a molecule so that properties such as formal charges or hybridisation can be averaged.